src/hexen/g_game.c

```diff
@@ -434,16 +461,30 @@ void G_BuildTiccmd(ticcmd_t *cmd, int maketic)
         gamekeydown[key_arti_teleportother] = false;
         cmd->arti = arti_teleportother;
     }
-    else if (gamekeydown[key_arti_egg] && !cmd->arti)
+    else if (gamekeydown[key_arti_poisonbag] && !cmd->arti)
     {
-        gamekeydown[key_arti_egg] = false;
-        cmd->arti = arti_egg;
+        gamekeydown[key_arti_poisonbag] = false;
+        cmd->arti = arti_poisonbag;
     }
-    else if (gamekeydown[key_arti_invulnerability] && !cmd->arti
-             && !players[consoleplayer].powers[pw_invulnerability])
+	else if (gamekeydown[key_arti_speed] && !cmd->arti)
     {
-        gamekeydown[key_arti_invulnerability] = false;
-        cmd->arti = arti_invulnerability;
+        gamekeydown[key_arti_speed] = false;
+        cmd->arti = arti_speed;
+    }
+	else if (gamekeydown[key_arti_boostmana] && !cmd->arti)
+    {
+        gamekeydown[key_arti_boostmana] = false;
+        cmd->arti = arti_boostmana;
+    }
+	else if (gamekeydown[key_arti_boostarmor] && !cmd->arti)
```

Comment:
Can you please expand any tab characters to spaces? As seen here, they will not appear correctly if the other person is not using your editor settings.

Reply:
Got it, will mess around with this some more, but unfortunately I don't think I'll be able to start working on it until the end of this week. Are there any reasons for which the Quartz Flask (health) and the Mystic Urn (superhealth) should **not** be grouped together? As far as I can see they (+ invuln) are the only ones that need separate checks, surprisingly. I was expecting healingradius (which has separate effects for each class) and boostarmor (maximum armor depends on class) to also need some class-based checks, but it appears that they are not necessary.